src/Simulation/Simulators.Tests/OperationsTestHelper.cs

```diff
@@ -156,11 +157,11 @@ internal static void ctrlTestShell(SimulatorBase sim, Action<(IQArray<Qubit>, Qu
                     // set control bits to match i:
                     for (int j = 0; j < n; j++)
                     {
-                        var value = (i & (1 << j)) == 0
-                            ? Result.Zero
-                            : Result.One;
-
-                        set.Apply((value, ctrls[j]));
+                        reset.__Body__(ctrls[j]);
+                        if ((i & (1 << j)) != 0)
+                        {
+                            x.__Body__(ctrls[j]);
```

Comment:
Yes, exactly. This was done to minimize dependency on `SetToBasisState` because of some complications that introduced in how I was building the target packages. Because `SetToBasisState` uses measurement, I like the idea of using `Reset` explicitly which may be implemented as `M` and conditional `X` or as an intrinsic dependending on the taget package getting tested. So while those compliciations have since been resolved and the code could be put back, I preferred this `Reset` based approach, personally.